src/triangulate/VoronoiDiagramBuilder.cpp

```diff
@@ -138,7 +138,15 @@ VoronoiDiagramBuilder::getDiagramEdges(const geom::GeometryFactory& geomFact)
     }
     std::unique_ptr<geom::Geometry> clipPoly(geomFact.toGeometry(&diagramEnv));
     std::unique_ptr<Geometry> clipped(clipPoly->intersection(edges.get()));
-    return clipped;
+
+    if (dynamic_cast<LineString*>(clipped.get())) {
```

Comment:
This looks like it will work, but it would be nice to avoid all the reaching into the `unique_ptr<>` quite so much.
```c++
if (clipped->getGeometryTypeId() == GEOS_LINESTRING) {
  std::vector<std::unique_ptr<LineString>> lines;
  lines.emplace_back(clipped.release());
  std::unique_ptr<MultiLineString> multi = geomFact.createMultiLineString(std::move(lines));
  return multi;
} else {
  return clipped;
}
```
This hasn't been subjected to a compiler or anything, but I think it will work?

Reply:
Done, thanks for the idea!